Nos/Controller/NoteEditorController.swift

```diff
@@ -0,0 +1,256 @@
+    
+    /// Inserts the name of an author at the current cursor position with a nostr link attached as an attribute.
+    func insertMention(of author: Author) {
+        guard let textView else { return }
+        self.insertMention(of: author, at: textView.selectedRange) 
+    }
+    
+    /// Appends the given string at the end of the text the user has entered.
+    func append(text: String) {
+        guard let textView else {
+            return
+        }
+        
+        let range = NSRange(location: textView.attributedText.length, length: 0)
+        let appendedAttributedString = NSAttributedString(
+            string: text,
+            attributes: defaultNSAttributes
+        )
+        
+        let attributedString = textView.attributedText.mutableCopy() as! NSMutableAttributedString
+        attributedString.replaceCharacters(in: range, with: appendedAttributedString)
+        textView.attributedText = attributedString
+        textView.selectedRange.location += appendedAttributedString.length
+    }
+    
+    /// Appends the given URL and adds the default link styling attributes. Will append a space before the link 
+    /// if needed.
+    func append(_ url: URL) {
+        guard let text = textView?.attributedText else { return }
+        
+        if let lastCharacter = text.string.last, !lastCharacter.isWhitespace {
+            append(text: " ")
+        }
+        
+        let range = NSRange(location: text.length, length: 0)
+        insert(text: url.absoluteString, link: url.absoluteString, at: range)
+    }
+    
+    // MARK: - UITextViewDelegate
+    
+    func textViewDidChange(_ textView: UITextView) {
+        updateIntrinsicHeight(view: textView)
+        isEmpty = textView.attributedText.length == 0
+    }
+        
+    func textView(
+        _ textView: UITextView, 
+        primaryActionFor textItem: UITextItem, 
+        defaultAction: UIAction
+    ) -> UIAction? {
+        nil
+    }
+    
+    func textView(
+        _ textView: UITextView,
+        shouldChangeTextIn nsRange: NSRange,
+        replacementText text: String
+    ) -> Bool {
+        let handledChange = removeLinkAttributesUnderCursor(selectedRange: nsRange, in: textView, newText: text)
+        if handledChange {
+            return false
+        }
+        textView.typingAttributes = defaultNSAttributes
+        
+        if text == "@" {
+            let showedAutocomplete = checkForMentionsAutocomplete(in: textView, at: nsRange)
+            if showedAutocomplete {
+                return true
+            }
+        } else if text.count > 1 {
+            let insertedNostrLink = handleNostrIdentifiers(in: text, textView: textView)
+            if insertedNostrLink {
+                return false
+            }
+        } 
+        
+        return true
+    }
+
+    // MARK: - Helpers 
+    
+    /// Calculates the height that fits all entered text, and updates $intrinsicHeight with this property. This is 
```

Comment:
Maybe $intrinsicHeight is technically accurate and if so, I guess we should leave this as-is. I'd prefer this, though:
```suggestion
    /// Calculates the height that fits all entered text, and updates `intrinsicHeight` with this property. This is 
```